src/update_flash.c

```diff
@@ -480,15 +494,17 @@ static int wolfBoot_delta_update(struct wolfBoot_image *boot,
         sector++;
     }
 #ifndef DISABLE_BACKUP
-    /* start re-entrant final erase, return code is only for resumption in
-     * wolfBoot_start */
-    wolfBoot_swap_and_final_erase(0);
 #endif
 out:
 #ifdef EXT_FLASH
     ext_flash_lock();
 #endif
     hal_flash_lock();
+    /* start re-entrant final erase, return code is only for resumption in
+     * wolfBoot_start */
+    if (ret == 0) {
+        wolfBoot_swap_and_final_erase(0);
+    }
```

Comment:
I think this should still be under `#ifndef DISABLE_BACKUP`, unless it's needed anyway. If that's the case, remove the empty `#ifndef` clause at L 496 then.

Reply:
missed that, fixed now